### ISSUE TITLE
Does not compile with 6.3 beta 4

### ISSUE THREAD

**Issue by dcharbonnier** [owensd/json-swift#31]
```
diff --git a/src/JSValue.Parsing.swift b/src/JSValue.Parsing.swift
index e9cdd8e..26dac32 100644
--- a/src/JSValue.Parsing.swift
+++ b/src/JSValue.Parsing.swift
@@ -268,7 +268,7 @@ extension JSValue {
                 fallthrough

             case (_, Token.Zero...Token.Nine, NumberParsingState.ExponentDigits):
-                exponent = exponent * 10 + codeunit - Token.Zero
+                exponent = exponent * 10 + Int(codeunit) - Int(Token.Zero)

             case (_, Token.Period, NumberParsingState.Whole):
                 state = .Decimal
@@ -383,11 +383,11 @@ extension JSValue {

     private static func parseHexDigit(digit: UInt8) -> Int? {
         if Token.Zero <= digit && digit <= Token.Nine {
-            return digit - Token.Zero
+            return Int(digit - Token.Zero)
         } else if Token.a <= digit && digit <= Token.f {
-            return 10 + digit - Token.a
+            return Int(10 + digit - Token.a)
         } else if Token.A <= digit && digit <= Token.F {
-            return 10 + digit - Token.A
+            return Int(10 + digit - Token.A)
         } else {
             return nil
         }
```

**Comment by dcharbonnier** [comment]
Working on fixing the tests but still have a problem
![image](https://cloud.githubusercontent.com/assets/6220422/7167808/54d0dcb2-e36d-11e4-820c-54efe45d3545.png)

**Comment by owensd** [comment]
Fixed with change: 6d4a3b7
